### ISSUE TITLE
panic : AGP needs intel_gtt_insert page see d6473f5 upstream in torvalds tree ........

### ISSUE THREAD

**Issue by javashin** [FreeBSDDesktop/freebsd-base-graphics#125]
FreeBSD darkmaza 12.0-CURRENT FreeBSD 12.0-CURRENT #30 a7f7424(drm-next): Tue Feb 21 23:42:39 UTC 2017     root@gauntlet:/usr/obj/usr/src/sys/GENERIC  amd64

this is with intel ironlake

**Comment by Juanitou** [comment]
I’m getting the same panic in a GM45 (mobile, Cantiga) which was working as for cd1bbc3. I’m using a recent 945ea51 and upgraded X packages.

**Comment by hselasky** [comment]
Does applying Torvalds d6473f5 solve the problem?

**Comment by mattmacy** [comment]
No we need a corresponding implementation of that function. It was at that change that the need arose.

**Comment by mattmacy** [comment]
I glossed over it in the 4.8 update.

**Comment by javashin** [comment]
no support for agp ironlake in the near future then ?

**Comment by mattmacy** [comment]
I don't think any of us can test. Maybe ask @dumbbell

**Comment by barulheira** [comment]
The same here: Intel Corporation 5 Series/3400 Series Chipset
(https://www.msi.com/Laptop/CR420MX.html)
Any hope?

**Comment by mattmacy** [comment]
Yes. I just have no access to test hardware right now.

**Comment by barulheira** [comment]
Perhaps I could help you, if you send me the binaries, or directions. I'm on TrueOS stable branch (drm-next).

**Comment by javashin** [comment]
no luck ?

**Comment by javashin** [comment]
??????

**Comment by mattmacy** [comment]
@markj please look at this when you get a chance. Thanks.

**Comment by javashin** [comment]
cool some attention

**Comment by markjdb** [comment]
Please test this commit: https://github.com/markjdb/freebsd-base-graphics/commit/c9032830aa7f92a824e05d7057fa985db9c0eccf

I don't have hardware to exercise it, so it's only compile-tested so far.

**Comment by javashin** [comment]
THANKS !!! A LOT 
is working and modesetting works with glamor and dri3 and uxa with dri3 too 
but with modesetting i get FIFO A ERROR UNDERRUN in the dmesg 
same as linux no problem there but i dont know if can be fixed ...
t
SNA WITH DRI3  and without dri3 i get artifacts glitches .all over the place. 
im upgrading to last git of the intel ddx which have a lot of fixes for sna 
cheers :+1: 
Thanks You. i think you can merge that commit to drm-next master i need to track freebsd head because of my wifi its uses the if_iwm driver . and adrian its making some upgrades progress to the iwm /

![screenshot at 2017-03-27 01-10-23](https://cloud.githubusercontent.com/assets/15115613/24344051/a246021e-1297-11e7-9ca4-9ed4ebc0466d.png)

**Comment by javashin** [comment]
Well So far everything is stable and nice :+1: 

FreeBSD darkmaza 12.0-CURRENT FreeBSD 12.0-CURRENT #0: Mon Mar 27 01:04:07 AST 2017     root@darkmaza:/usr/obj/usr/src/drm-next/sys/BIOSHOCK  amd64


System:    Host: darkmaza Kernel: 12.0-CURRENT amd64 (64 bit) Desktop: MATE 1.18.0  OS: FreeBSD 12.0-CURRENT
Machine:   System: TOSHIBA product: PORTEGE R700 v: PT310U-0K002Y01 serial: 3B053928H
           Mobo: TOSHIBA model: Portable PC serial: 0000000000
           Bios: TOSHIBA v: Version 2.20 rv 2.20 date: 12/22/2011
CPU:       Cores: 4 model: Intel Core i3 M 380 (-SMP-) cache: N/A speed/max: 2527/2533 MHz
Graphics:  Card: Intel Core Processor Integrated Graphics Controller
           Display Server: X.Org 1.19.3 driver: intel Resolution: 1366x768@60.17hz
           GLX Renderer: Mesa DRI Intel Ironlake Mobile GLX Version: 2.1 Mesa 13.0.5
Audio:     Card Intel 5 Series/3400 Series High Definition Audio driver: hdac
Network:   Card: Intel Wireless 7260 driver: iwm
           IF: iwm0 state: N/A mac: N/A
Drives:    HDD Total Size: 305.2GB (1.4% used) ID-1: ada0 model: N/A size: 305.2GB
Partition: ID-1: / size: 83G used: 23G (31%) fs: ufs dev: /dev/ada0s4a
           ID-2: swap-1 size: 4.26GB used: 0.00GB (0%) fs: swap dev: /dev/ada0s4b
RAID:      Device-1: tank - - raid: zfs-no-raid components: online: 9640772848720761907UNAVAIL
Sensors:   This feature is not yet supported for BSD systems.
Info:      Processes: 107 Uptime: 32 mins Memory: 2630.1/3804.1MB Client: Shell (bash) inxi: 2.2.31 

Copyright (c) 1992-2017 The FreeBSD Project.
Copyright (c) 1979, 1980, 1983, 1986, 1988, 1989, 1991, 1992, 1993, 1994
	The Regents of the University of California. All rights reserved.
FreeBSD is a registered trademark of The FreeBSD Foundation.
FreeBSD 12.0-CURRENT #0: Mon Mar 27 01:04:07 AST 2017
    root@darkmaza:/usr/obj/usr/src/drm-next/sys/BIOSHOCK amd64
FreeBSD clang version 4.0.0 (tags/RELEASE_400/final 297347) (based on LLVM 4.0.0)
VT(vga): resolution 640x480
CPU: Intel(R) Core(TM) i3 CPU       M 380  @ 2.53GHz (2527.06-MHz K8-class CPU)
  Origin="GenuineIntel"  Id=0x20655  Family=0x6  Model=0x25  Stepping=5
  Features=0xbfebfbff<FPU,VME,DE,PSE,TSC,MSR,PAE,MCE,CX8,APIC,SEP,MTRR,PGE,MCA,CMOV,PAT,PSE36,CLFLUSH,DTS,ACPI,MMX,FXSR,SSE,SSE2,SS,HTT,TM,PBE>
  Features2=0x9ae3bd<SSE3,DTES64,MON,DS_CPL,VMX,EST,TM2,SSSE3,CX16,xTPR,PDCM,PCID,SSE4.1,SSE4.2,POPCNT>
  AMD Features=0x28100800<SYSCALL,NX,RDTSCP,LM>
  AMD Features2=0x1<LAHF>
  VT-x: PAT,HLT,MTF,PAUSE,EPT,UG,VPID
  TSC: P-state invariant, performance statistics
real memory  = 4294967296 (4096 MB)
avail memory = 3825094656 (3647 MB)
Event timer "LAPIC" quality 600
ACPI APIC Table: <TOSHIB A007B   >
FreeBSD/SMP: Multiprocessor System Detected: 4 CPUs
FreeBSD/SMP: 1 package(s) x 2 core(s) x 2 hardware threads
random: unblocking device.
ioapic0: Changing APIC ID to 1
ioapic0 <Version 2.0> irqs 0-23 on motherboard
SMP: AP CPU #1 Launched!
SMP: AP CPU #3 Launched!
SMP: AP CPU #2 Launched!
Timecounter "TSC-low" frequency 1263530648 Hz quality 1000
Cuse v0.1.34 @ /dev/cuse
random: entropy device external interface
netmap: loaded module
kbd1 at kbdmux0
nexus0
vtvga0: <VT VGA driver> on motherboard
cryptosoft0: <software crypto> on motherboard
acpi0: <TOSHIB A007B> on motherboard
acpi0: Power Button (fixed)
hpet0: <High Precision Event Timer> iomem 0xfed00000-0xfed003ff on acpi0
Timecounter "HPET" frequency 14318180 Hz quality 950
Event timer "HPET" frequency 14318180 Hz quality 550
Event timer "HPET1" frequency 14318180 Hz quality 440
Event timer "HPET2" frequency 14318180 Hz quality 440
Event timer "HPET3" frequency 14318180 Hz quality 440
Event timer "HPET4" frequency 14318180 Hz quality 440
cpu0: <ACPI CPU> on acpi0
cpu1: <ACPI CPU> on acpi0
cpu2: <ACPI CPU> on acpi0
cpu3: <ACPI CPU> on acpi0
atrtc0: <AT realtime clock> port 0x70-0x77 irq 8 on acpi0
atrtc0: Warning: Couldn't map I/O.
Event timer "RTC" frequency 32768 Hz quality 0
attimer0: <AT timer> port 0x40-0x43,0x50-0x53 irq 0 on acpi0
Timecounter "i8254" frequency 1193182 Hz quality 0
Event timer "i8254" frequency 1193182 Hz quality 100
Timecounter "ACPI-fast" frequency 3579545 Hz quality 900
acpi_timer0: <24-bit timer at 3.579545MHz> port 0x408-0x40b on acpi0
pcib0: <ACPI Host-PCI bridge> port 0xcf8-0xcff on acpi0
pcib0: could not evaluate _ADR - AE_NOT_FOUND
pci0: <ACPI PCI bus> on pcib0
vgapci0: <VGA-compatible display> port 0x3030-0x3037 mem 0xd0000000-0xd03fffff,0xc0000000-0xcfffffff irq 16 at device 2.0 on pci0
agp0: <Intel Ironlake (M) SVGA controller> on vgapci0
agp0: aperture size is 256M, detected 65532k stolen memory
vgapci0: Boot video device
pci0: <simple comms> at device 22.0 (no driver attached)
ehci0: <Intel PCH USB 2.0 controller USB-B> mem 0xd4709000-0xd47093ff irq 16 at device 26.0 on pci0
usbus0: EHCI version 1.0
usbus0 on ehci0
usbus0: 480Mbps High Speed USB v2.0
hdac0: <Intel 5 Series/3400 Series HDA Controller> mem 0xd4700000-0xd4703fff irq 22 at device 27.0 on pci0
pcib1: <ACPI PCI-PCI bridge> irq 16 at device 28.0 on pci0
pci1: <ACPI PCI bus> on pcib1
sdhci_pci0: <RICOH R5CE822 SD> mem 0xd4600000-0xd46000ff irq 16 at device 0.0 on pci1
sdhci_pci0: 1 slot(s) allocated
pcib2: <ACPI PCI-PCI bridge> irq 17 at device 28.1 on pci0
pci2: <ACPI PCI bus> on pcib2
iwm0: <Intel(R) Dual Band Wireless AC 7260> mem 0xd4500000-0xd4501fff irq 17 at device 0.0 on pci2
pcib3: <ACPI PCI-PCI bridge> irq 18 at device 28.2 on pci0
pcib3: failed to allocate initial prefetch window: 0xd0400000-0xd23fffff
pci3: <ACPI PCI bus> on pcib3
pcib4: <ACPI PCI-PCI bridge> irq 19 at device 28.3 on pci0
pci4: <ACPI PCI bus> on pcib4
ehci1: <Intel PCH USB 2.0 controller USB-A> mem 0xd4708000-0xd47083ff irq 23 at device 29.0 on pci0
usbus1: EHCI version 1.0
usbus1 on ehci1
usbus1: 480Mbps High Speed USB v2.0
pcib5: <ACPI PCI-PCI bridge> at device 30.0 on pci0
pci5: <ACPI PCI bus> on pcib5
isab0: <PCI-ISA bridge> at device 31.0 on pci0
isa0: <ISA bus> on isab0
ahci0: <Intel 5 Series/3400 Series AHCI SATA controller> port 0x3028-0x302f,0x303c-0x303f,0x3020-0x3027,0x3038-0x303b,0x3000-0x301f mem 0xd4707000-0xd47077ff irq 19 at device 31.2 on pci0
ahci0: AHCI v1.30 with 4 3Gbps ports, Port Multiplier not supported
ahcich0: <AHCI channel> at channel 0 on ahci0
ahcich1: <AHCI channel> at channel 1 on ahci0
ahciem0: <AHCI enclosure management bridge> on ahci0
pcib6: <ACPI Host-PCI bridge> on acpi0
pci6: <ACPI PCI bus> on pcib6
acpi_acad0: <AC Adapter> on acpi0
acpi_button0: <Power Button> on acpi0
acpi_lid0: <Control Method Lid Switch> on acpi0
battery0: <ACPI Control Method Battery> on acpi0
acpi_tz0: <Thermal Zone> on acpi0
atkbdc0: <Keyboard controller (i8042)> port 0x60,0x64 irq 1 on acpi0
atkbd0: <AT Keyboard> irq 1 on atkbdc0
kbd0 at atkbd0
atkbd0: [GIANT-LOCKED]
psm0: <PS/2 Mouse> irq 12 on atkbdc0
psm0: [GIANT-LOCKED]
psm0: model Synaptics Touchpad, device ID 0
orm0: <ISA Option ROM> at iomem 0xe0000-0xeffff on isa0
coretemp0: <CPU On-Die Thermal Sensors> on cpu0
est0: <Enhanced SpeedStep Frequency Control> on cpu0
coretemp1: <CPU On-Die Thermal Sensors> on cpu1
est1: <Enhanced SpeedStep Frequency Control> on cpu1
coretemp2: <CPU On-Die Thermal Sensors> on cpu2
est2: <Enhanced SpeedStep Frequency Control> on cpu2
coretemp3: <CPU On-Die Thermal Sensors> on cpu3
est3: <Enhanced SpeedStep Frequency Control> on cpu3
fuse-freebsd: version 0.4.4, FUSE ABI 7.8
ZFS filesystem version: 5
ZFS storage pool version: features support (5000)
Timecounters tick every 1.000 msec
hdacc0: <Realtek ALC269 HDA CODEC> at cad 0 on hdac0
hdaa0: <Realtek ALC269 Audio Function Group> at nid 1 on hdacc0
pcm0: <Realtek ALC269 (Internal Analog)> at nid 20 and 18 on hdaa0
pcm1: <Realtek ALC269 (Right Analog)> at nid 33 and 24 on hdaa0
hdacc1: <Intel Ibex Peak HDA CODEC> at cad 3 on hdac0
hdaa1: <Intel Ibex Peak Audio Function Group> at nid 1 on hdacc1
pcm2: <Intel Ibex Peak (HDMI/DP 8ch)> at nid 4 on hdaa1
pcm3: <Intel Ibex Peak (HDMI/DP 8ch)> at nid 5 on hdaa1
ugen1.1: <Intel EHCI root HUB> at usbus1
ugen0.1: <Intel EHCI root HUB> at usbus0
uhub0: ses0 at ahciem0 bus 0 scbus2 target 0 lun 0
<Intel EHCI root HUB, class 9/0, rev 2.00/1.00, addr 1> on usbus1
ses0: <AHCI SGPIO Enclosure 1.00 0001> SEMB S-E-S 2.00 device
uhub1: ses0: SEMB SES Device
<Intel EHCI root HUB, class 9/0, rev 2.00/1.00, addr 1> on usbus0
ada0 at ahcich0 bus 0 scbus0 target 0 lun 0
iwm0: ada0: <WDC WD3200BEKT-60F3T1 12.01A12> ATA8-ACS SATA 2.x device
ada0: Serial Number WD-WX20AA9K3948
hw rev 0x140, fw ver 17.352738.0, address f0:42:1c:c7:07:59
ada0: 300.000MB/s transfers (SATA 2.x, UDMA5, PIO 8192bytes)
ada0: Command Queueing enabled
ada0: 305245MB (625142448 512 byte sectors)
Steering write from 0 kBps to 300000 kBps
cd0 at ahcich1 bus 0 scbus1 target 0 lun 0
cd0: <MATSHITA DVD-RAM UJ892ES 1.20> Removable CD-ROM SCSI device
cd0: Serial Number HK31 305776
cd0: 150.000MB/s transfers (SATA 1.x, UDMA5, ATAPI 12bytes, PIO 8192bytes)
cd0: Attempt to query device size failed: NOT READY, Medium not present - tray closed
Trying to mount root from ufs:/dev/ada0s4a []...
uhub0: 2 ports with 2 removable, self powered
uhub1: 2 ports with 2 removable, self powered
ugen1.2: <vendor 0x8087 product 0x0020> at usbus1
uhub2 on uhub0
uhub2: <vendor 0x8087 product 0x0020, class 9/0, rev 2.00/0.00, addr 2> on usbus1
ugen0.2: <vendor 0x8087 product 0x0020> at usbus0
uhub3 on uhub1
uhub3: <vendor 0x8087 product 0x0020, class 9/0, rev 2.00/0.00, addr 2> on usbus0
acpi_video0: <ACPI video extension> on vgapci0
uhub3: 6 ports with 6 removable, self powered
uhub2: 8 ports with 8 removable, self powered
wlan0: Ethernet address: f0:42:1c:c7:07:59
ugen0.3: <vendor 0x08ff Fingerprint Sensor> at usbus0
ugen1.3: <vendor 0x8087 product 0x07dc> at usbus1
ugen0.4: <Realtek 2SF001> at usbus0
iwm0: iwm_update_edca: called
iwm0: iwm_update_edca: called
wlan0: link state changed to UP
[drm] Initialized
drmn0: <drmn> on vgapci0
vgapci0: child drmn0 requested pci_enable_io
vgapci0: child drmn0 requested pci_enable_io
[drm] Memory usable by graphics device = 2048M
[drm] Supports vblank timestamp caching Rev 2 (21.10.2013).
[drm] Driver supports precise vblank timestamp query.
[drm] Connector LVDS-1: get mode from tunables:
[drm]   - kern.vt.fb.modes.LVDS-1
[drm]   - kern.vt.fb.default_mode
[drm] Connector VGA-1: get mode from tunables:
[drm]   - kern.vt.fb.modes.VGA-1
[drm]   - kern.vt.fb.default_mode
[drm] Connector HDMI-A-1: get mode from tunables:
[drm]   - kern.vt.fb.modes.HDMI-A-1
[drm]   - kern.vt.fb.default_mode
[drm] Connector DP-1: get mode from tunables:
[drm]   - kern.vt.fb.modes.DP-1
[drm]   - kern.vt.fb.default_mode
[drm] Connector HDMI-A-2: get mode from tunables:
[drm]   - kern.vt.fb.modes.HDMI-A-2
[drm]   - kern.vt.fb.default_mode
[drm] Connector DP-2: get mode from tunables:
[drm]   - kern.vt.fb.modes.DP-2
[drm]   - kern.vt.fb.default_mode
[drm] RC6 disabled, disabling runtime PM support
[drm] Initialized i915 1.6.0 20160919 for drmn on minor 0
VT: Replacing driver "vga" with new "fb".
start FB_INFO:
type=11 height=768 width=1366 depth=32
cmsize=16 size=4227072
pbase=0xc0046000 vbase=0xfffff800c0046000
name=drmn0 flags=0x0 stride=5504 bpp=32
cmap[0]=0 cmap[1]=7f0000 cmap[2]=7f00 cmap[3]=c4a000
end FB_INFO
drmn0: fb0: inteldrmfb frame buffer device



    28.582] 
X.Org X Server 1.19.3
Release Date: 2017-03-15
[    28.582] X Protocol Version 11, Revision 0
[    28.582] Build Operating System: FreeBSD 12.0-CURRENT amd64 
[    28.582] Current Operating System: FreeBSD darkmaza 12.0-CURRENT FreeBSD 12.0-CURRENT #0: Mon Mar 27 01:04:07 AST 2017     root@darkmaza:/usr/obj/usr/src/drm-next/sys/BIOSHOCK amd64
[    28.582] Build Date: 24 March 2017  05:37:42AM
[    28.582]  
[    28.582] Current version of pixman: 0.34.0
[    28.582] 	Before reporting problems, check http://wiki.x.org
	to make sure that you have the latest version.
[    28.582] Markers: (--) probed, (**) from config file, (==) default setting,
	(++) from command line, (!!) notice, (II) informational,
	(WW) warning, (EE) error, (NI) not implemented, (??) unknown.
[    28.583] (==) Log file: "/var/log/Xorg.0.log", Time: Mon Mar 27 02:20:10 2017
[    28.703] (==) Using config file: "/etc/X11/xorg.conf"
[    28.703] (==) Using system config directory "/usr/local/share/X11/xorg.conf.d"
[    28.746] (==) ServerLayout "X.org Configured"
[    28.746] (**) |-->Screen "Screen0" (0)
[    28.746] (**) |   |-->Monitor "Monitor0"
[    28.844] (**) |   |-->Device "Card0"
[    28.844] (**) |-->Input Device "Mouse0"
[    28.844] (**) |-->Input Device "Keyboard0"
[    28.844] (==) Not automatically adding devices
[    28.844] (==) Not automatically enabling devices
[    28.844] (==) Not automatically adding GPU devices
[    28.844] (==) Max clients allowed: 256, resource mask: 0x1fffff
[    29.035] (**) FontPath set to:
	/usr/local/share/fonts/misc/,
	/usr/local/share/fonts/TTF/,
	/usr/local/share/fonts/OTF/,
	/usr/local/share/fonts/Type1/,
	/usr/local/share/fonts/100dpi/,
	/usr/local/share/fonts/75dpi/,
	/usr/local/share/fonts/misc/,
	/usr/local/share/fonts/TTF/,
	/usr/local/share/fonts/OTF/,
	/usr/local/share/fonts/Type1/,
	/usr/local/share/fonts/100dpi/,
	/usr/local/share/fonts/75dpi/
[    29.036] (**) ModulePath set to "/usr/local/lib/xorg/modules"
[    29.036] (**) Extension "Composite" is enabled
[    29.036] (II) Loader magic: 0x817e30
[    29.036] (II) Module ABI versions:
[    29.036] 	X.Org ANSI C Emulation: 0.4
[    29.036] 	X.Org Video Driver: 23.0
[    29.036] 	X.Org XInput driver : 24.1
[    29.036] 	X.Org Server Extension : 10.0
[    29.036] (--) PCI:*(0:0:2:0) 8086:0046:1179:0007 rev 2, Mem @ 0xd0000000/4194304, 0xc0000000/268435456, I/O @ 0x00003030/8, BIOS @ 0x????????/65536
[    29.036] (II) "glx" will be loaded. This was enabled by default and also specified in the config file.
[    29.036] (II) LoadModule: "glx"
[    29.154] (II) Loading /usr/local/lib/xorg/modules/extensions/libglx.so
[    29.303] (II) Module glx: vendor="X.Org Foundation"
[    29.303] 	compiled for 1.19.3, module version = 1.0.0
[    29.303] 	ABI class: X.Org Server Extension, version 10.0
[    29.303] (II) LoadModule: "intel"
[    29.385] (II) Loading /usr/local/lib/xorg/modules/drivers/intel_drv.so
[    29.521] (II) Module intel: vendor="X.Org Foundation"
[    29.521] 	compiled for 1.19.2, module version = 2.99.917
[    29.521] 	Module class: X.Org Video Driver
[    29.521] 	ABI class: X.Org Video Driver, version 23.0
[    29.521] (II) LoadModule: "mouse"
[    29.521] (II) Loading /usr/local/lib/xorg/modules/input/mouse_drv.so
[    29.545] (II) Module mouse: vendor="X.Org Foundation"
[    29.545] 	compiled for 1.19.2, module version = 1.9.2
[    29.545] 	Module class: X.Org XInput Driver
[    29.545] 	ABI class: X.Org XInput driver, version 24.1
[    29.545] (II) LoadModule: "kbd"
[    29.545] (II) Loading /usr/local/lib/xorg/modules/input/kbd_drv.so
[    29.546] (II) Module kbd: vendor="X.Org Foundation"
[    29.546] 	compiled for 1.19.2, module version = 1.9.0
[    29.546] 	Module class: X.Org XInput Driver
[    29.546] 	ABI class: X.Org XInput driver, version 24.1
[    29.546] (II) intel: Driver for Intel(R) Integrated Graphics Chipsets:
	i810, i810-dc100, i810e, i815, i830M, 845G, 854, 852GM/855GM, 865G,
	915G, E7221 (i915), 915GM, 945G, 945GM, 945GME, Pineview GM,
	Pineview G, 965G, G35, 965Q, 946GZ, 965GM, 965GME/GLE, G33, Q35, Q33,
	GM45, 4 Series, G45/G43, Q45/Q43, G41, B43
[    29.547] (II) intel: Driver for Intel(R) HD Graphics
[    29.547] (II) intel: Driver for Intel(R) Iris(TM) Graphics
[    29.547] (II) intel: Driver for Intel(R) Iris(TM) Pro Graphics
[    29.547] (--) Using syscons driver with X support (version 2.0)
[    29.547] (++) using VT number 9

[    30.634] (WW) VGA arbiter: cannot open kernel arbiter, no multi-card support
[    30.634] (==) intel(0): Depth 24, (--) framebuffer bpp 32
[    30.634] (==) intel(0): RGB weight 888
[    30.634] (==) intel(0): Default visual is TrueColor
[    30.634] (**) intel(0): Option "AccelMethod" "UXA"
[    30.634] (**) intel(0): Option "Backlight" "intel_backlight"
[    30.634] (**) intel(0): Option "DRI" "3"
[    30.634] (**) intel(0): Option "XvMC" "true"
[    30.634] (--) intel(0): Integrated Graphics Chipset: Intel(R) HD Graphics
[    30.647] (**) intel(0): Relaxed fencing enabled
[    30.647] (**) intel(0): Wait on SwapBuffers? enabled
[    30.647] (**) intel(0): Triple buffering? enabled
[    30.647] (**) intel(0): Framebuffer tiled
[    30.647] (**) intel(0): Pixmaps tiled
[    30.647] (**) intel(0): 3D buffers tiled
[    30.647] (**) intel(0): SwapBuffers wait enabled
[    30.647] (==) intel(0): video overlay key set to 0x101fe
[    31.047] (II) intel(0): Output LVDS1 using monitor section Monitor0
[    31.082] (EE) intel(0): unrecognised backlight control interface intel_backlight
[    31.087] (II) intel(0): Output VGA1 has no monitor section
[    31.095] (II) intel(0): Output HDMI1 has no monitor section
[    31.095] (II) intel(0): Output DP1 has no monitor section
[    31.103] (II) intel(0): Output HDMI2 has no monitor section
[    31.103] (II) intel(0): Output DP2 has no monitor section
[    31.114] (II) intel(0): EDID for output LVDS1
[    31.116] (II) intel(0): Not using default mode "320x240" (doublescan mode not supported)
[    31.116] (II) intel(0): Not using default mode "400x300" (doublescan mode not supported)
[    31.116] (II) intel(0): Not using default mode "400x300" (doublescan mode not supported)
[    31.116] (II) intel(0): Not using default mode "512x384" (doublescan mode not supported)
[    31.116] (II) intel(0): Not using default mode "640x480" (doublescan mode not supported)
[    31.116] (II) intel(0): Not using default mode "640x512" (doublescan mode not supported)
[    31.116] (II) intel(0): Not using default mode "800x600" (doublescan mode not supported)
[    31.116] (II) intel(0): Not using default mode "896x672" (doublescan mode not supported)
[    31.116] (II) intel(0): Not using default mode "928x696" (doublescan mode not supported)
[    31.116] (II) intel(0): Not using default mode "960x720" (doublescan mode not supported)
[    31.116] (II) intel(0): Not using default mode "700x525" (doublescan mode not supported)
[    31.116] (II) intel(0): Not using default mode "1024x768" (doublescan mode not supported)
[    31.116] (II) intel(0): Printing probed modes for output LVDS1
[    31.116] (II) intel(0): Modeline "1366x768"x60.2   76.22  1366 1400 1512 1624  768 769 771 780 -hsync -vsync (46.9 kHz P)
[    31.116] (II) intel(0): Modeline "1024x768"x60.0   65.00  1024 1048 1184 1344  768 771 777 806 -hsync -vsync (48.4 kHz d)
[    31.116] (II) intel(0): Modeline "800x600"x60.3   40.00  800 840 968 1056  600 601 605 628 +hsync +vsync (37.9 kHz d)
[    31.116] (II) intel(0): Modeline "800x600"x56.2   36.00  800 824 896 1024  600 601 603 625 +hsync +vsync (35.2 kHz d)
[    31.116] (II) intel(0): Modeline "640x480"x59.9   25.18  640 656 752 800  480 490 492 525 -hsync -vsync (31.5 kHz d)
[    31.121] (II) intel(0): EDID for output VGA1
[    31.129] (II) intel(0): EDID for output HDMI1
[    31.129] (II) intel(0): EDID for output DP1
[    31.137] (II) intel(0): EDID for output HDMI2
[    31.137] (II) intel(0): EDID for output DP2
[    31.137] (II) intel(0): Output LVDS1 connected
[    31.137] (II) intel(0): Output VGA1 disconnected
[    31.137] (II) intel(0): Output HDMI1 disconnected
[    31.137] (II) intel(0): Output DP1 disconnected
[    31.137] (II) intel(0): Output HDMI2 disconnected
[    31.137] (II) intel(0): Output DP2 disconnected
[    31.137] (II) intel(0): Using exact sizes for initial modes
[    31.137] (II) intel(0): Output LVDS1 using initial mode 1366x768 +0+0
[    31.137] (II) intel(0): Kernel page flipping support detected, enabling
[    31.137] (==) intel(0): Using gamma correction (1.0, 1.0, 1.0)
[    31.137] (==) intel(0): DPI set to (96, 96)
[    31.137] (II) Loading sub module "fb"
[    31.137] (II) LoadModule: "fb"
[    31.137] (II) Loading /usr/local/lib/xorg/modules/libfb.so
[    31.151] (II) Module fb: vendor="X.Org Foundation"
[    31.151] 	compiled for 1.19.3, module version = 1.0.0
[    31.151] 	ABI class: X.Org ANSI C Emulation, version 0.4
[    31.151] (II) Loading sub module "dri2"
[    31.151] (II) LoadModule: "dri2"
[    31.151] (II) Module "dri2" already built-in
[    31.151] (II) Loading sub module "dri3"
[    31.151] (II) LoadModule: "dri3"
[    31.151] (II) Module "dri3" already built-in
[    31.151] (==) Depth 24 pixmap format is 32 bpp
[    31.151] (II) intel(0): Allocated new frame buffer 1366x768 stride 5632, tiled
[    31.182] (II) UXA(0): Driver registered support for the following operations:
[    31.182] (II)         solid
[    31.182] (II)         copy
[    31.182] (II)         composite (RENDER acceleration)
[    31.182] (II)         put_image
[    31.182] (II)         get_image
[    31.182] (II) intel(0): [DRI2] Setup complete
[    31.182] (II) intel(0): [DRI2]   DRI driver: i965
[    31.182] (II) intel(0): [DRI2]   VDPAU driver: va_gl
[    31.182] (==) intel(0): Backing store enabled
[    31.182] (==) intel(0): Silken mouse enabled
[    31.183] (II) intel(0): Initializing HW Cursor
[    31.190] (II) intel(0): RandR 1.2 enabled, ignore the following RandR disabled message.
[    31.191] (==) intel(0): DPMS enabled
[    31.191] (**) intel(0): Intel XvMC decoder enabled
[    31.192] (II) intel(0): Set up textured video
[    31.192] (II) intel(0): [XvMC] xvmc_vld driver initialized.
[    31.192] (II) intel(0): DRI2: Enabled
[    31.192] (II) intel(0): DRI3: Enabled
[    31.231] (--) RandR disabled
[    31.449] (II) AIGLX: enabled GLX_MESA_copy_sub_buffer
[    31.450] (II) AIGLX: enabled GLX_ARB_create_context
[    31.450] (II) AIGLX: enabled GLX_ARB_create_context_profile
[    31.450] (II) AIGLX: enabled GLX_EXT_create_context_es{,2}_profile
[    31.450] (II) AIGLX: enabled GLX_INTEL_swap_event
[    31.450] (II) AIGLX: enabled GLX_SGI_swap_control
[    31.450] (II) AIGLX: enabled GLX_EXT_framebuffer_sRGB
[    31.450] (II) AIGLX: enabled GLX_ARB_fbconfig_float
[    31.450] (II) AIGLX: enabled GLX_EXT_fbconfig_packed_float
[    31.450] (II) AIGLX: GLX_EXT_texture_from_pixmap backed by buffer objects
[    31.450] (II) AIGLX: Loaded and initialized i965
[    31.450] (II) GLX: Initialized DRI2 GL provider for screen 0
[    31.450] (II) intel(0): Setting screen physical size to 361 x 203
[    32.053] (II) Using input driver 'mouse' for 'Mouse0'
[    32.053] (**) Option "CorePointer"
[    32.053] (**) Mouse0: always reports core events
[    32.053] (**) Option "Protocol" "auto"
[    32.053] (**) Option "Device" "/dev/sysmouse"
[    32.053] (**) Mouse0: Protocol: "auto"
[    32.053] (**) Mouse0: always reports core events
[    32.053] (==) Mouse0: Emulate3Buttons, Emulate3Timeout: 50
[    32.053] (**) Option "ZAxisMapping" "4 5 6 7"
[    32.054] (**) Mouse0: ZAxisMapping: buttons 4, 5, 6 and 7
[    32.054] (**) Mouse0: Buttons: 7
[    32.054] (II) XINPUT: Adding extended input device "Mouse0" (type: MOUSE, id 6)
[    32.054] (**) Mouse0: (accel) keeping acceleration scheme 1
[    32.054] (**) Mouse0: (accel) acceleration profile 0
[    32.054] (**) Mouse0: (accel) acceleration factor: 2.000
[    32.054] (**) Mouse0: (accel) acceleration threshold: 4
[    32.054] (II) Mouse0: SetupAuto: hw.iftype is 4, hw.model is 0
[    32.054] (II) Mouse0: SetupAuto: protocol is SysMouse
[    32.054] (II) Using input driver 'kbd' for 'Keyboard0'
[    32.054] (**) Option "CoreKeyboard"
[    32.054] (**) Keyboard0: always reports core events
[    32.054] (**) Keyboard0: always reports core events
[    32.054] (**) Option "Protocol" "standard"
[    32.054] (**) Option "XkbRules" "base"
[    32.054] (**) Option "XkbModel" "pc105"
[    32.054] (**) Option "XkbLayout" "us"
[    32.054] (II) XINPUT: Adding extended input device "Keyboard0" (type: KEYBOARD, id 7)

**Comment by javashin** [comment]
![screenshot at 2017-03-27 02-57-39](https://cloud.githubusercontent.com/assets/15115613/24344538/f614fc36-1299-11e7-8ec1-6b3076d99000.png)

**Comment by markjdb** [comment]
Thanks for testing. To be clear, you're still seeing artifacts with SNA configured?

**Comment by javashin** [comment]
Sadly yes i still get graphics corruption with sna with the intel ddx from git master ,
and with modesetting i get FIFO A UNDERUN error like i get on linux with the last kernel 4.10.X.
with UXA + DRI3 everything is fine .

**Comment by Juanitou** [comment]
Issue solved here with a GM45 (mobile, Cantiga) in an old laptop. No artifacts with SNA and DRI3 enabled through xorg.conf. Thanks!

**Comment by markjdb** [comment]
Ok. I'm going to close this issue; the corruption issue is most likely unrelated and deserves its own issue.
